src/playfair.rs

```diff
@@ -0,0 +1,429 @@
+/// bottom side of the column)."
+///
+/// [Reference](https://en.wikipedia.org/wiki/Playfair_cipher#Description)
+fn apply_row_col(b: &Bigram, row_col: &[String; 5], mode: &CipherMode) -> Option<Bigram> {
+    for rc in row_col.iter() {
+        if let Some(first) = rc.find(b.0) {
+            if let Some(second) = rc.find(b.1) {
+                let v: Vec<char> = rc.chars().collect();
+                match *mode {
+                    CipherMode::ENCRYPT => return Some((v[(first + 1) % 5], v[(second + 1) % 5])),
+                    CipherMode::DECRYPT => return Some((v[(first - 1) % 5], v[(second - 1) % 5])),
+                }
+            }
+        }
+    }
+    None
+}
+
+/// Identifies 2 corners of the rectangle.
+fn find_separate(b: &Bigram, table: &[String; 5]) -> (usize, usize) {
+    let mut result = (0, 0);
+    for rc in table.iter() {
+        if let Some(pos) = rc.find(b.0) {
+            result.0 = pos;
+            continue;
+        }
+        if let Some(pos) = rc.find(b.1) {
+            result.1 = pos;
+            continue;
+        }
+    }
+    result
+}
+
+/// Apply rule 4 (Rectangle).
+///
+/// "If the letters are not on the same row or column, replace them with
+/// the letters on the same row respectively but at the other pair of
+/// corners of the rectangle defined by the original pair. The order is
+/// important – the first letter of the encrypted pair is the one that
+/// lies on the same row as the first letter of the plaintext pair."
+///
+/// [Reference](https://en.wikipedia.org/wiki/Playfair_cipher#Description)
+fn apply_rectangle(b: &Bigram, table: &KeyTable) -> Bigram {
+    let rows = find_separate(&b, &table.cols);
+    let cols = find_separate(&b, &table.rows);
+
+    let row0: Vec<char> = table.rows[rows.0].chars().collect();
+    let row1: Vec<char> = table.rows[rows.1].chars().collect();
+
+    (row0[cols.1], row1[cols.0])
+}
+
+/// Apply the PlayFair cipher algorithm.
+///
+/// The operations for encrypt and decrypt are identical
+/// except for the "direction" of the substitution choice.
+fn apply_rules(
+    bigrams: Vec<Bigram>,
+    table: &KeyTable,
+    mode: CipherMode,
+) -> Result<String, &'static str> {
+    let mut text = String::new();
+    for b in bigrams {
+        // Rule 2 (Row)
+        if let Some(bigram) = apply_row_col(&b, &table.rows, &mode) {
+            text.push(bigram.0);
+            text.push(bigram.1);
+            continue;
+        }
+
+        // Rule 3 (Column)
+        if let Some(bigram) = apply_row_col(&b, &table.cols, &mode) {
+            text.push(bigram.0);
+            text.push(bigram.1);
+            continue;
+        }
+
+        // Rule 4 (Rectangle)
+        let bigram = apply_rectangle(&b, &table);
+        text.push(bigram.0);
+        text.push(bigram.1);
+    }
+    Ok(text)
+}
+
+/// A 5x5 Playfair key table
+struct KeyTable {
+    /// Table rows
+    rows: [String; 5],
+    /// Table columns
+    cols: [String; 5],
+}
+
+/// A Playfair cipher.
```

Comment:
Could you ensure that your documentation style is in keeping with the other Ciphers